app/models/record_edit.rb

```diff
@@ -0,0 +1,27 @@
+class RecordEdit < ActiveRecord::Base
+
+  belongs_to :record, polymorphic: true
+  belongs_to :author, class_name: 'User'
+  belongs_to :user, foreign_key: 'author_id' #duplicate author relationship for eager loading via discussion.items
+  has_one    :event, as: :eventable
+
+  validates_presence_of :author, :record
+
+  def self.capture(record, author)
```

Comment:
PaperTrail supports a 'changeset' hash property with a minor migration out of the box, and I don't think it's a great pick at this point to rip it out. Was there an attempt at using the original versioning stuff there that didn't work for some reason?